File: ruby/mercuryledger/tests/unit/workflow_test.rb
# frozen_string_literal: true

require_relative '../test_helper'

class WorkflowTest < Minitest::Test
  def test_transition_graph_enforced
    assert MercuryLedger::Core::Workflow.transition_allowed?(:queued, :allocated)
    refute MercuryLedger::Core::Workflow.transition_allowed?(:queued, :arrived)
  end
end

File: js/signaldock/tests/unit/workflow.test.js
const test = require('node:test');
const assert = require('node:assert/strict');
const { canTransition } = require('../../src/core/workflow');

test('workflow transitions remain valid', () => {
  assert.equal(canTransition('queued', 'allocated'), true);
  assert.equal(canTransition('queued', 'arrived'), false);
});
